testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/permissionset/AddPermissionSetOnlineTest.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    @After
+    public void cleanup() throws Exception {
+        ops.removeIfExists(TEST_PERMISSION_SET_ADDRESS);
+        ops.removeIfExists(TEST_PERMISSION_SET_ADDRESS2);
+        administration.reloadIfRequired();
+    }
+
+    @Test
+    public void addPermissionSet() throws Exception {
+        AddPermissionSet addPermissionSet
+                = new AddPermissionSet.Builder(TEST_PERMISSION_SET_NAME)
+                .build();
+
+        client.apply(addPermissionSet);
+
+        assertTrue("Permission set should be created", ops.exists(TEST_PERMISSION_SET_ADDRESS));
+    }
+
+    @Test
+    public void addTwoPermissionSets() throws Exception {
+        AddPermissionSet addPermissionSet
+                = new AddPermissionSet.Builder(TEST_PERMISSION_SET_NAME)
+                .build();
+
+        AddPermissionSet addPermissionSet2
+                = new AddPermissionSet.Builder(TEST_PERMISSION_SET_NAME2)
+                .build();
+
+        client.apply(addPermissionSet);
+        client.apply(addPermissionSet2);
+
+        assertTrue("Permission set should be created", ops.exists(TEST_PERMISSION_SET_ADDRESS));
+        assertTrue("Second Permission set should be created", ops.exists(TEST_PERMISSION_SET_ADDRESS2));
+    }
+
+    @Test
+    public void addFullPermissionSet() throws Exception {
+        AddPermissionSet addPermissionSet
+                = new AddPermissionSet.Builder(TEST_PERMISSION_SET_NAME)
+                .addPermissions(new AddPermissionSet.PermissionBuilder()
+                        .className("org.wildfly.security.auth.permission.LoginPermission")
+                        .action("login")
+                        .targetName("loginPermissionName")
+                        .build(),
+                        new AddPermissionSet.PermissionBuilder()
+                        .className("org.wildfly.security.auth.permission.RunAsPrincipalPermission")
+                        .action("read")
+                        .targetName("runAsPrincipalPermissionName")
+                        .build())
+                .build();
+
+        client.apply(addPermissionSet);
+
+        assertTrue("Permission set should be created", ops.exists(TEST_PERMISSION_SET_ADDRESS));
+
+        checkPermissionSetAttribute("permissions[0].class-name",
+                "org.wildfly.security.auth.permission.LoginPermission");
+        checkPermissionSetAttribute("permissions[0].action", "login");
+        checkPermissionSetAttribute("permissions[0].target-name", "loginPermissionName");
+
+        checkPermissionSetAttribute("permissions[1].class-name",
+                "org.wildfly.security.auth.permission.RunAsPrincipalPermission");
+        checkPermissionSetAttribute("permissions[1].action", "read");
+        checkPermissionSetAttribute("permissions[1].target-name", "runAsPrincipalPermissionName");
+    }
+
+    @Test(expected = CommandFailedException.class)
+    public void addExistPermissionSetNotAllowed() throws Exception {
```

Comment:
Can you add test where in one set 2 permissions with same attribute values are added. Now it should be possible.